ldpd compile failures w/ clang and -werror

ldpe.c:146:12: error: address of array 'init->user' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
        if (init->user)
        ~~  ~~~~~~^~~~
ldpe.c:148:12: error: address of array 'init->group' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
        if (init->group)
        ~~  ~~~~~~^~~~~
lde.c:167:12: error: address of array 'init->user' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
        if (init->user)
        ~~  ~~~~~~^~~~
lde.c:169:12: error: address of array 'init->group' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
        if (init->group)
        ~~  ~~~~~~^~~~~

--- a/ldpd/lde.c
+++ b/ldpd/lde.c
@@ -80,10 +80,6 @@ static zebra_capabilities_t _caps_p [] =
 
 static struct zebra_privs_t lde_privs =
 {
-#if defined(FRR_USER) && defined(FRR_GROUP)
-	.user = FRR_USER,
-	.group = FRR_GROUP,
-#endif
 #if defined(VTY_GROUP)
 	.vty_group = VTY_GROUP,
 #endif
@@ -164,10 +160,8 @@ void
 lde_init(struct ldpd_init *init)
 {
 	/* drop privileges */
-	if (init->user)
-		lde_privs.user = init->user;
-	if (init->group)
-		lde_privs.group = init->group;
+	lde_privs.user = init->user;
+	lde_privs.group = init->group;
 	zprivs_init(&lde_privs);
 
 #ifdef HAVE_PLEDGE

--- a/ldpd/ldpe.c
+++ b/ldpd/ldpe.c
@@ -66,10 +66,6 @@ static zebra_capabilities_t _caps_p [] =
 
 struct zebra_privs_t ldpe_privs =
 {
-#if defined(FRR_USER) && defined(FRR_GROUP)
-	.user = FRR_USER,
-	.group = FRR_GROUP,
-#endif
 #if defined(VTY_GROUP)
 	.vty_group = VTY_GROUP,
 #endif
@@ -143,10 +139,8 @@ void
 ldpe_init(struct ldpd_init *init)
 {
 	/* drop privileges */
-	if (init->user)
-		ldpe_privs.user = init->user;
-	if (init->group)
-		ldpe_privs.group = init->group;
+	ldpe_privs.user = init->user;
+	ldpe_privs.group = init->group;
 	zprivs_init(&ldpe_privs);
 
 	/* listen on ldpd control socket */

